app/src/main/java/io/homeassistant/companion/android/notifications/NotificationActionReceiver.kt

```diff
@@ -0,0 +1,38 @@
+package io.homeassistant.companion.android.notifications
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class NotificationActionReceiver : BroadcastReceiver() {
+
+    companion object {
+        const val FIRE_EVENT = "FIRE_EVENT"
+        const val EXTRA_NOTIFICATION_ACTION = "EXTRA_ACTION_KEY"
+    }
+
+    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + Job())
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationUseCase
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        DaggerServiceComponent.builder()
+            .appComponent((context!!.applicationContext as GraphComponentAccessor).appComponent)
+            .build()
+            .inject(this)
+
+        intent?.getParcelableExtra<NotificationAction>(EXTRA_NOTIFICATION_ACTION)?.let {
+            ioScope.launch {
+                integrationUseCase.fireEvent(it.key, it.data)
```

Comment:
We should not allow the user to pick the event name to fire. Instead, we should pick an event name like "mobile_app_action" and pass both the key and data as event data.

Reply:
Actually, why do we include `it.data` ? We should drop that.

Reply:
We are basically ecoing back the data that the user passed in the notification.  You can put any string->string mapping in there.  We could limit it to a standard event name and only pass the key if you want.  I was just trying to leave it more open, I think the iOS app echos all data back too.

Reply:
ok

Reply:
Which option would you like?

Reply:
I updated to use constant event type and added an "action" element to the event data that is the selected option.

Reply:
I'm fine with echo'ing it back.